### PR TITLE
Implement available periods

### LINKED ISSUE #80: Support Available Periods
https://github.com/cronofy/cronofy-ruby/pull/78

### PR DESCRIPTION
**Pull request by jeremyw**
Adds the following methods to `Cronofy::Client` with specs:

- `#upsert_available_period(available_period_id:, start:, end:, tzid:)`
- `#get_available_periods`
- `#delete_available_period(id)`

Fixes #80

### CODE REVIEW

File: lib/cronofy/client.rb [hunk 1]
```diff
@@ -1492,6 +1492,69 @@ def delete_availability_rule(availability_rule_id)
...
+    # Raises Cronofy::CredentialsMissingError if no credentials available.
+    # Raises Cronofy::InvalidRequestError if the request contains invalid
+    # parameters.
+    # Raises Cronofy::TooManyRequestsError if the request exceeds the rate
+    # limits for the application.
+    def get_available_periods(options={})
+      query = [
+        "from=#{to_iso8601(options[:from])}",
+        "to=#{to_iso8601(options[:to])}",
+        "tzid=#{options[:tzid]}"
```
Comment (dzajic) [inline lib/cronofy/client.rb:1534]: Looks like there's also a "localized_times" option? https://docs.cronofy.com/developers/api/scheduling/available-periods/read/#param-localized_times

Comment (jeremyw) [reply lib/cronofy/client.rb:1534]: We aren't using that parameter. Support could be added in a separate PR.

**Comment by jeremyw** [comment]
@gshutler FYI - any feedback?

**Comment by AdamWhittingham** [comment]
Thank you for this contribution @jeremyw!

I'm going to close this PR and #81 will add support for the available periods end point.

**Pull request closed without merging**